scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -888,14 +1006,51 @@ private ClickHandler clickHandlerForSubmitButton() {
         return new ClickHandler() {
             @Override
             public void onClick(ClickEvent event) {
-                for (int i = 0; i < _fields.length; i++) {
-                    String val = "";
-                    if (fields[2 * i].getValue() != null) {
-                        val = fields[2 * i].getValue().toString();
+                setAllValuesAsString();
+                enableValidation(false);
+                handleRadioButton();
+                variablesActualForm.addSubmitCompleteHandler(new SubmitCompleteHandler() {
+                    @Override
+                    public void onSubmitComplete(SubmitCompleteEvent event) {
+                        handleJobSubmission(event);
                     }
-                    _fields[i].setValue(val);
+
+                    private void handleJobSubmission(SubmitCompleteEvent event) {
+                        checkEventResults(event);
+                    }
+
+                });
+                variablesActualForm.submit();
+                displayLoadingMessage();
+
+            }
+
+            private void checkEventResults(SubmitCompleteEvent event) {
+                if (event.getResults() == null || !event.getResults().startsWith(SubmitEditServlet.ERROR)) {
+
+                    GWT.log("Job submitted to the scheduler");
+                    SubmitWindow.this.window.removeMember(rootPage);
+                    SubmitWindow.this.window.hide();
+                    SubmitWindow.this.destroy();
+                } else {
+                    String jsonError = event.getResults().substring(SubmitEditServlet.ERROR.length());
+                    JSONValue json = controller.parseJSON(jsonError);
+                    JSONObject obj = json.isObject();
+                    if (obj != null && obj.containsKey(ERROR_MESSAGE)) {
+                        String errorMessage = obj.get(ERROR_MESSAGE).toString();
+                        if (errorMessage.contains("JobValidationException")) {
+                            errorMessage = errorMessage.substring(errorMessage.indexOf(":") + 2);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Put single-quotes around ':' to use the faster "indexOf(char)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3027)